File: ruby/primes.rb
# primes.rb -- Ruby
# ruby primes.rb
class Primes
    def generate_primes(total)
        $primes = []
        $primes[0] = 2
        x = 1
        k = 1
        while (k < total)
            x += 2
            j = 0
            while (j < k && ((x % $primes[j]) != 0))
                if ($primes[j] * $primes[j] >= x)
                    $primes[k] = x
                    k += 1
                    break
                end
                j += 1
            end
        end
        return $primes
    end
end

p = Primes.new
total = 10000
printf("素数表 Ruby\n")

start_time = Time.now
$primes = p.generate_primes(total)
end_time = Time.now
interval = end_time - start_time
printf("計算時間：%.8f秒\n", interval)

start_time = Time.now
for i in 0...total
    if (i % 10 == 0); printf("\n"); end
    printf("%8d", $primes[i])
end
end_time = Time.now

printf("\n")
interval = end_time - start_time
printf("出力時間：%.8f秒\n", interval)
exit 0

File: python/primes.py
# coding=utf-8
# primes.py -- 素数（Python）
# python primes.py
import time

class Primes():
    def generate_primes(self, total) :
        primes = [0] * total
        primes[0] = 2
        x = 1
        k = 1
        while (k < total):
            x += 2
            j = 0;
            while (j < k and x % primes[j] != 0):
                if (primes[j] * primes[j] >= x):
                    primes[k] = x
                    k = k + 1
                    break
                j = j + 1

        return primes

print("素数表 Python")
p = Primes()
total = 10000

start_time = time.perf_counter()
primes = p.generate_primes(total)
end_time = time.perf_counter()
interval = (end_time - start_time)
print("")
print("計算時間：{0}秒".format(interval))

start_time = time.perf_counter()
for i in range(0, total):
    if (i % 10 == 0):
        print("")
    print("{:<8}".format(primes[i]), end="")

end_time = time.perf_counter()
interval = (end_time - start_time)
print("")
print("出力時間：{0}秒".format(interval))